src/revisions_parser.py

```diff
@@ -2,37 +2,120 @@
 from typing import Dict, List
 
 
-def parse(input: str) -> List[Dict[str, str]]:
-    # find start
-    iter = re.finditer(r"revision history|version control", input,
-                       re.IGNORECASE)
-    start = None
-    a = 0
-    for start in iter:
-        a += 1
-        # first mention probably in toc
-        if a == 2:
-            break
-    if not start:
-        return []
-    input = input[start.span(0)[1]:]
+def month_to_number(input: str) -> str:
+    values = {
+        "jan": "01",
+        "feb": "02",
+        "mar": "03",
+        "apr": "04",
+        "may": "05",
+        "jun": "06",
+        "jul": "07",
+        "aug": "08",
+        "sep": "09",
+        "oct": "10",
+        "nov": "11",
+        "dec": "12",
+    }
+    form = input[:3].lower()
+    if form in values:
+        return values[form]
+    return ""
+
+
+def clean_date(input: str) -> str:
+    if "-" in input:
+        splitted = input.split("-")
+        if len(splitted) != 3:
+            return input
+
+        if not splitted[1].isnumeric():
+            # swap year and day
+            tmp = splitted[0]
+            splitted[0] = splitted[2]
+            splitted[2] = tmp
 
-    # find end
+            # use number for months
+            splitted[1] = month_to_number(splitted[1])
+    elif "." in input:
+        splitted = input.split(".")
+        if len(splitted) != 3:
+            return input
+
+        # swap year and day
+        tmp = splitted[0]
+        splitted[0] = splitted[2]
+        splitted[2] = tmp
+    else:
+        return input
+
+    return "-".join(splitted)
+
+
+def parse_general(input: str, regex: str, ver_index: int, date_index: int) -> List[Dict[str, str]]:
     end = re.search(r"\n{4}", input, re.IGNORECASE)
     if end:
         input = input[:end.span(0)[1]]
     else:
         input = input[:5000]
-    
-    result = re.findall(r"\s+([0-9.]+)\s+([0-9-]+)?\s+(.*)", input)
-    
+
+    result = re.findall(regex, input)
+
+    final_result = []
     for i in range(len(result)):
-        result[i] = {
-            "version": result[i][0],
-            "date": result[i][1],
-            # there can be more information, like author...
-            "description": result[i][2].split("  ")[-1],
-        }
+        version = result[i][ver_index]
+        date = clean_date(result[i][date_index])
+        # there can be more information, like author...
+        description = result[i][2].split("  ")[-1]
+
+        final_result.append({
+            "version": version,
+            "date": date,
+            "description": description,
+        })
+    return final_result
+
+
+def parse_rev_date_desc(input: str) -> List[Dict[str, str]]:
+    return parse_general(input, r"\s+v?([0-9.]+)\s+([0-9-A-Za-z-\.]+)?[\s:]\s+(.*)", 0, 1)
+
+
+def parse_date_ver_desc(input: str) -> List[Dict[str, str]]:
+    return parse_general(input, r"\s+v?([0-9-A-Za-z-\.]+)?\s+([0-9.]+)[\s:]\s+(.*)", 1, 0)
```

Comment:
The use of `-` in the subregex `[0-9-A-Za-z-\.]` seems like a misuse of the fact that you (probably) wanted to make the character `-` to be a part of the class. As you can surely imagine, it is not the best option for maintenance.